main.js

```diff
@@ -620,6 +662,256 @@ app.post('/delete-admin', async function(req, res) {
 	}
 });
 
+app.get('/kits', async (req, res) => {
+	if (redirectToLogin(req, res)) {
+		return;
+	}
+	const filterQuery = {'_id': new mdb.ObjectId(req.session._id)};
+	const user = await db.collection('BBY-6_users').findOne(filterQuery);
+	if(!user) {
+		// Could not find user
+		res.status(500).send("userNotFound");
+		return;
+	}
+	res.status(200).send(user.kits);
+})
+
+app.post('/kits', async (req, res) => {
+	if (redirectToLogin(req, res)) {
+		return;
+	}
+	try {
+		const filterQuery = {'name': req.body.name};
+		const kitTemplate = await db.collection('BBY-6_kit-templates').findOne(filterQuery);
+		if(!kitTemplate) {
+			res.status(500).send("kitTemplateNotFound");
+			return;
+		}
+		try {
+			const userFilterQuery = {'_id': new mdb.ObjectId(req.session._id)};
+			kitTemplate.kit = kitTemplate.kit.map(element => {
+				element.required = true;
+				element.completed = false;	
+				return element;
+			})
+			const user = await db.collection('BBY-6_users').updateOne(userFilterQuery, {$push: {kits: kitTemplate}});
+			if(!user) {
+				// Could not find user
+				res.status(500).send("userNotFound");
+				return;
+			}
+			res.status(200).send(kitTemplate);
+		} catch(e) {
+			res.status(500).send("serverIssue");
+		}
+	} catch(e) {
+		res.status(500).send("serverIssue");
+	}
+})
+
+app.patch('/kits', async (req, res) => {
+	if (redirectToLogin(req, res)) {
+		return;
+	}
+	if(!req.body || !req.body._id || !req.body.itemId || req.body.completed === undefined) {
+		res.status(400).send("missingBodyArguments");
+		return;
+	}
+	const filterQuery = {
+		'_id': new mdb.ObjectId(req.session._id),
+		'kits._id': new mdb.ObjectId(req.body._id),
+	}
+	try {
+		const result = await db.collection('BBY-6_users').updateOne(filterQuery, {$set: {"kits.$[i].kit.$[j].completed": req.body.completed}}, 
+			{
+				arrayFilters: [ 
+					{"i._id": new mdb.ObjectId(req.body._id)}, 
+					{"j._id": new mdb.ObjectId(req.body.itemId)}
+				]
+			}
+		);
+		if(!result) {
+			// Could not find user 
+			res.status(500).send("couldNotFindUser");
+			return;
+		} else if (result.matchedCount === 0) {
+			// Could not find kit
+			res.status(404).send("couldNotFindKit");
+			return;
+		}
+		res.status(200).send("kitUpdatedSuccessfully");
+	} catch(e) {
+		res.status(500).send("serverIssue");
+	}
+})
+
+app.delete('/kits', async (req, res) => {
+	if (redirectToLogin(req, res)) {
+		return;
+	}
+	if(!req.body || !req.body._id) {
+		res.status(400).send("missingBodyArguments");
+		return;
+	}
+	const filterQuery = {
+		'_id': new mdb.ObjectId(req.session._id),
+		'kits._id': new mdb.ObjectId(req.body._id)
+	}
+	try {
+		req.body._id = new mdb.ObjectId(req.body._id);
```

Comment:
```suggestion
```
Wait, what, don't you repeat that same `new mdb.ObjectId` constructor the next line down? 🤔

Reply:
Maybe it's error handling stuff (early exit because of try/catch??) – if so, it should be documented with a comment.